<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSI/iLogos/nsst.jpg" width="400" height="100">

# DEMI | NOVA SST

## Mecânica dos Sólidos I

## Tração e compressão de barras elásticas e elastoplásticas perfeitas
## Vigas sujeitas à tracção/compressão. Parte III
## Problema 5 <a name="PIV105"></a>

Uma barra de secção circular com 50 mm de diâmetro está fixa nas suas extremidades $A$ e $B$ como se mostra na figura. Na secção $C$ está aplicada uma força $P$ = 400 kN, alinhada com o eixo da barra e com o sentido indicado e, simultaneamente, a barra é aquecida 20º C. O material da barra pode ser caracterizado como tendo um comportamento elástico perfeitamente plástico, com o diagrama tensão-extensão indicado na figura e o coeficiente de dilatação térmica é $\alpha = 12\times 10{-6}$ /ºC.

[a)](#A) Determine as tensões normais que ocorrem nos dois troços, $AC$ e $CB$, da barra.

[b)](#B) Calcule o valor do deslocamento horizontal da secção $C$.

[c)](#C) Considere agora que a força $P$ era retirada e que a barra regressava à temperatura inicial e determine o valor da tensão residual que se verificaria na barra, assim como o deslocamento residual da secção $C$ em relação à posição inicial.

<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSI/Notebooks/Au10/Au10_P005_i1.jpg"
width="700">

In [4]:
import numpy as np
import matplotlib.pyplot as plt

plt.rcParams.update({'font.size': 22})
# use LaTeX fonts in the plot
# plt.rc('text', usetex=True)
plt.rc('font', family='serif')

# Dados:

# unidades: mm, N, MPa

LAC, LBC  = 600., 900.
L = LAC + LBC
d = 50.
A = np.pi*d**2/4
print(f'A (secção transversal) = {A:.1f} mm2')

Sced = 140.
Epsced = 0.00148
Fced = Sced*A
print(f'Força de cedência = {Fced:.2f} N')
E = Sced/Epsced
print(f'E = {E*1e-3:.2f} GPa')

# Carregamento térmico
DT = 20. # unidades: ºC
alpha = 12.e-6 # unidades: /ºC

# Carregamento mecânico
P = 400.e3 # unidade: N

A (secção transversal) = 1963.5 mm2
Força de cedência = 274889.36 N
E = 94.59 GPa


#### a) Determine as tensões normais que ocorrem nos dois troços, $AC$ e $CB$, da barra. <a id="A"></a>

([enunciatum](#PIV105))

- Forças devidas à variação de temperatura $\Delta T$

Por aplicação do princípio de sobreposição ao problema estaticamente indeterminado considere-se a barra encastrada numa extremidade e livre na outra (bordo livre por remoção de um encastramento). A viga estaticamente equivalente está submetida a uma variação de temperatura e a uma força redundante (a reação do apoio). Da equação de compatibilidade de deslocamentos, resulta a seguinte expressão:

\begin{equation*}
\delta^{\Delta T} = \delta^{F}
\quad\Leftrightarrow\quad
\frac{FL}{AE} = \alpha\Delta T L
\end{equation*}

In [5]:
Ft = alpha*DT*A*E
print(f'Força de cedência = {Ft:.2f} N')

Força de cedência = 44576.65 N


- Forças devidas à acção da força $P$

Pelo equilíbrio da barra, as forças de reação nas secções $A$ e $B$ devem igual a força aplicada. Considerando um comportamento elástico:

\begin{equation*}
F_A^e + F_B^e = P
\end{equation*}

Pela equação de compatabilidade entre as secções externas do veio: $\delta_{A/B} = 0$, ou seja,

\begin{equation*}
-\delta_{AC}^e + \delta_{BC}^e = 0
\quad\Leftrightarrow\quad
\frac{F_A^e L_{AC}}{EA} = \frac{F_B^e L_{BC}}{EA}
\quad\Leftrightarrow\quad
F_A^e = \frac{L_{BC}}{L_{AC}} F_B^e
\end{equation*}

onde $\delta_{AC}^e$  representa a diminuição de comprimento da barra $AC$ e  $\delta_{BC}^e$  representa o aumento de comprimento da barra $BC$, gerado pela força $P$ aplicada em $C$.

In [16]:
from sympy.solvers import solve
from sympy import symbols

RA, RB  = symbols('RA RB')

eq1 = RA + RB - P
eq2 = RA - LBC/LAC*RB
sol = solve([eq1,eq2], [RA, RB], dict=True)
FAe = float(sol[0][RA])
FBe = float(sol[0][RB])
print(f'FAe = {FAe*1e-3:.1f} kN')
print(f'FBe = {FBe*1e-3:.1f} kN')

FAe = 240.0 kN
FBe = 160.0 kN


In [17]:
FresA = Ft + FAe
print(f'FresA (elástico) = {FresA:.2f} N')
FresB = FBe - Ft
print(f'FresB (elástico) = {FresB:.2f} N')

FresA (elástico) = 284576.65 N
FresB (elástico) = 115423.35 N


Atendendo a que a força de cedência (correspondente à tensão de cedência) é igual a 274889.4 N, conclui-se que ocorre plastificação na barra $AC$.

De acordo com o modelo de comportamento elasto-plástico, a força máxima exercida no troço $AC$ é a força de cedência, pelo que, para o equilíbrio a força na reação na secção $B$ deverá garantir:

\begin{equation*}
F_B = P - F_A^\mathrm{ced}
\end{equation*}

In [18]:
FBp = P - Fced
print(f'FBp (regime plástico do troço AC) = {FBp:.2f} N')

FBp (regime plástico do troço AC) = 125110.64 N


Pelo que as tensões normais nas barras serão:

\begin{equation*}
\begin{cases}
\sigma_{AC} = \sigma_C \quad\textrm{(tensão normal de compressão)}\\
\sigma_{BC} = F_{B_P}/A \quad\textrm{(tensão normal de tração)}\\
\end{cases}
\end{equation*}

In [19]:
sigmaAC = -Sced
print(f'sigmaAC = {sigmaAC:.2f} MPa (compressão)')

sigmaBC = FBp/A
print(f'sigmaBC = {sigmaBC:.2f} MPa (tração)')

sigmaAC = -140.00 MPa (compressão)
sigmaBC = 63.72 MPa (tração)


<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSI/Notebooks/Au10/Au10_P005_i2.jpg"
width="600">

#### b) Calcule o valor do deslocamento horizontal da secção $C$. <a id="B"></a>

([enunciatum](#PIV105))

O deslocamento horizontal efetivo da secção $C$ é calculado considerando o princípio de sobreposição. Olhando para a secção $C$ pelo troço $BC$ com comportamento linear elástico vem:

\begin{equation*}
\left(\delta_{C} = \delta_{BC}^e + \delta_{BC}^{\Delta T}\right)_{BC}
\quad\Leftrightarrow\quad
\delta_{C} = \frac{F_{B_P}L_{BC}}{EA} + \alpha\Delta T L_{BC}
\end{equation*}

In [20]:
deltaCBCe = FBp*LBC/E/A
print(f'( delta C^e )BC = {deltaCBCe:.3f} mm')

deltaCBCt = alpha*DT*LBC
print(f'( delta C^t )BC = {deltaCBCt:.3f} mm')

deltaCBC = deltaCBCe + deltaCBCt
print(f'( delta C )BC = {deltaCBC:.3f} mm (alongamento da barra BC)')

( delta C^e )BC = 0.606 mm
( delta C^t )BC = 0.216 mm
( delta C )BC = 0.822 mm (alongamento da barra BC)


#### c) Considere agora que a força $P$ era retirada e que a barra regressava à temperatura inicial e determine o valor da tensão residual que se verificaria na barra, assim como o deslocamento residual da secção $C$ em relação à posição inicial. <a id="C"></a>

([enunciatum](#PIV105))

Pela equação de compatibilidade de deslocamentos pode-se escrever:

\begin{equation*}
\begin{aligned}
&\delta_{AC}^\mathrm{total} + \delta_{AC}^\mathrm{total} = 0 \\
\quad\Leftrightarrow\quad &7
\delta_{AC}^{\Delta T} - \delta_{AC}^{F}  + \delta_{BC}^{\Delta T} + \delta_{BC}^{F} = 0 \\
\quad\Leftrightarrow\quad &
\delta_{AC}^{\Delta T} - \delta_{AC}^{e} - \delta_{AC}^{p} + \delta_{BC}^{\Delta T} + \delta_{BC}^{e} = 0 \\
\quad\Leftrightarrow\quad &
\delta_{AC}^{p} = \delta_{AC}^{\Delta T} - \delta_{AC}^{e}  + \delta_{BC}^{\Delta T} + \delta_{BC}^{e} \\
\end{aligned}
\end{equation*}

<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSI/Notebooks/Au10/Au10_P005_i3.jpg"
width="600">

In [21]:
deltaACterm = alpha*DT*LAC
print(f'( delta AC ) térmico = {deltaACterm:.5f} mm (alongamento térmico da barra AC)7')
deltaACe = Fced*LAC/E/A
print(f'( delta AC ) elástico = {deltaACe:.5f} mm (encurtamento elástico da barra AC)')
deltaABplas = deltaACterm - deltaACe + deltaCBC
print(f'( delta AC ) plástico = {deltaABplas:.5f} mm (encurtamento plástico da barra AC)')

( delta AC ) térmico = 0.14400 mm (alongamento térmico da barra AC)7
( delta AC ) elástico = 0.88800 mm (encurtamento elástico da barra AC)
( delta AC ) plástico = 0.07823 mm (encurtamento plástico da barra AC)


A deformação permanente plástica no troço $AC$ irá gerar uma tensão residual no barra quando a força $P$ e o aumento de temperatura forem retirados.

O delocamento residual, $\delta_{AC}^{p}$, irá ser proporcional à força residual pela expressão:

\begin{equation*}
\left(\delta_R = \frac{F_R L}{AE}\right)_{AB}
\end{equation*}

Obtida a força residual a tensão residual é obtida por:

\begin{equation*}
\sigma^\mathrm{res} = \frac{F_R}{A}
\end{equation*}

In [22]:
Fresidual = deltaABplas*A*E/L
print(f'Força residual = {Fresidual:.2f} N')
sigmaresidual = Fresidual/A
print(f'Tensão residual = {sigmaresidual:.2f} MPa')

Força residual = 9687.30 N
Tensão residual = 4.93 MPa


A secção $C$ desloca-se para a esquerda:

\begin{equation*}
\left(\delta_C^R = \frac{\sigma_R L}{E}\right)_{BC}
\end{equation*}

In [23]:
desltaCR = sigmaresidual*LBC/E
print(f'( delta C ) residual = {desltaCR:.5f} mm')

( delta C ) residual = 0.04694 mm


---

Copyright (c) [Mecânica dos Sólidos I - DEMI - NOVA-SST]

Interactive computing by <a href="https://jupyter.org/" target="_blank"> <span
style="color:#333399"> Jupyter Notebook </span> </a> &nbsp;|&nbsp;Coded by <a href = "mailto: jmc.xavier@fct.unl.pt">José Xavier</a>

Licensed under  <a href="http://creativecommons.org/licenses/by-sa/4.0/"
target="_blank"> <span style="color:#333399;font-size: 20px"> CC BY-SA 4.0  </span></a>
